lib/faraday/request/authorization.rb

```diff
@@ -39,16 +41,53 @@ def self.build_hash(type, hash)
 
       # @param app [#call]
       # @param type [String, Symbol] Type of Authorization
-      # @param token [String, Symbol, Hash] Token value for the Authorization
-      def initialize(app, type, token)
-        @header_value = self.class.header(type, token)
+      # @param params [Array<String, Proc>] parameters to build the Authorization header.
+      #   If the type is `:basic`, then these can be a login and password pair.
+      #   Otherwise, a single value is expected that will be appended after the type.
+      #   This value can be a proc, in which case it will be invoked on each request.
+      def initialize(app, type, *params)
+        @type = type
+        @params = params
+        @header_value = self.class.header(type, params[0]) unless params[0].is_a? Proc
         super(app)
       end
 
       # @param env [Faraday::Env]
-      def call(env)
-        env.request_headers[KEY] = @header_value unless env.request_headers[KEY]
-        @app.call(env)
+      # def call(env)
+      #   env.request_headers[KEY] = @header_value unless env.request_headers[KEY]
+      #   @app.call(env)
+      # end
+      #
```

Comment:
Perhaps this will fix the Yard junk error?

```suggestion

```

Reply:
Ah yes, YARD thinks this is a single comment for the `on_request` method.
@jarl-dk It doesn't make sense to leave the old `call` method commented, we can simply get rid of it.

Reply:
I agree that was a mistake, I'll remove it.